tests/eko/test_ad_as1aem1.py

```diff
@@ -32,11 +35,74 @@ def test_photon_momentum_conservation():
     # photon momentum
     N = complex(2.0, 0.0)
     sx = get_sx(N)
-    # import pdb; pdb.set_trace()
     np.testing.assert_almost_equal(
-        +2 * NU * constants.eu2 * as1aem1.gamma_qph(N, NF, sx)
-        + 2 * ND * constants.ed2 * as1aem1.gamma_qph(N, NF, sx)
-        + (NU * constants.eu2 + ND * constants.ed2) * as1aem1.gamma_phph()
-        + (NU * constants.eu2 + ND * constants.ed2) * as1aem1.gamma_gph(N),
+        constants.eu2 * ad.as1aem1.gamma_qph(N, NU, sx)
+        + constants.ed2 * ad.as1aem1.gamma_qph(N, ND, sx)
+        + (NU * constants.eu2 + ND * constants.ed2) * ad.as1aem1.gamma_phph()
+        + (NU * constants.eu2 + ND * constants.ed2) * ad.as1aem1.gamma_gph(N),
+        0,
+    )
+
+
+def test_quark_momentum_conservation():
+    # quark momentum
+    N = complex(2.0, 0.0)
+    sx = get_sx(N)
+    np.testing.assert_almost_equal(
+        ad.as1aem1.gamma_nsp(N, sx)
+        + ad.as1aem1.gamma_gq(N, sx)
+        + ad.as1aem1.gamma_phq(N, sx),
+        0,
+        decimal=4,
+    )
+
+
+NF = 6
```

Comment:
this will overwrite the previous definition and hence you only test nf=6 - instead you should test both nf=5 and nf=6, but this you need to iterate *inside* the function (remember that python functions get interpreted upon calling); furthermore you define twice `test_number_conservation` (as it does not depend on nf)

here is a suggestion what you can do instead:
```py
def test_gluon_momentum_conservation():
  N = complex(2.0, 0.0)
  sx = get_sx(N)
  for nf, nd in ((5,3),(6,3)):
    nu = nf - nd
    # actual test here depending on nf,nd,nu
```